File: test_data/code/java/1437-check-if-all-1s-are-at-least-length-k-places-away.java
class Solution {
    public boolean kLengthApart(int[] nums, int k) {
        for (int i = 0; i < nums.length - 1; i++) {
            if (nums[i] == 1 && i + 1 < k) {
                return false;
            }
            if (nums[i] == 1 && i + 1 >= k) {
                i += k - 1;
            }
        }
        return true;
    }
}

File: test_data/code/python/1437-check-if-all-1s-are-at-least-length-k-places-away.py
class Solution:
    def kLengthApart(self, nums: List[int], k: int) -> bool:
        prev_one = -1
        for i, num in enumerate(nums):
            if num == 1:
                if prev_one != -1 and i - prev_one <= k:
                    return False
                prev_one = i
        return True